integration/hive/src/test/java/org/apache/carbondata/hive/CarbonHiveRecordReaderTest.java

```diff
@@ -0,0 +1,234 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.hive;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
+import org.apache.carbondata.common.CarbonIterator;
+import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier;
+import org.apache.carbondata.core.metadata.CarbonTableIdentifier;
+import org.apache.carbondata.core.metadata.ColumnarFormatVersion;
+import org.apache.carbondata.core.scan.executor.exception.QueryExecutionException;
+import org.apache.carbondata.core.scan.executor.impl.DetailQueryExecutor;
+import org.apache.carbondata.core.scan.executor.infos.BlockExecutionInfo;
+import org.apache.carbondata.core.scan.model.QueryModel;
+import org.apache.carbondata.core.scan.result.BatchResult;
+import org.apache.carbondata.core.scan.result.iterator.AbstractDetailQueryResultIterator;
+import org.apache.carbondata.core.scan.result.iterator.ChunkRowIterator;
+import org.apache.carbondata.core.scan.result.iterator.DetailQueryResultIterator;
+import org.apache.carbondata.hadoop.readsupport.CarbonReadSupport;
+
+import mockit.Mock;
+import mockit.MockUp;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CarbonHiveRecordReaderTest {
+    private static CarbonHiveRecordReader carbonHiveRecordReaderObj;
+    private static AbsoluteTableIdentifier absoluteTableIdentifier;
+    private static QueryModel queryModel = new QueryModel();
+    private static CarbonReadSupport<ArrayWritable> readSupport = new CarbonDictionaryDecodeReadSupport<>();
+    private static JobConf jobConf = new JobConf();
+    private static InputSplit inputSplitNotInstanceOfHiveInputSplit, inputSplitInstanceOfHiveInputSplit;
+    private static BatchResult batchResult = new BatchResult();
+    private static Writable writable;
+    private static CarbonIterator carbonIteratorObject;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        String array[] = {"neha", "01", "vaishali"};
+        writable = new ArrayWritable(array);
+        absoluteTableIdentifier = new AbsoluteTableIdentifier(
+                "carbondata/examples/spark2/target/store",
+                new CarbonTableIdentifier("DB", "TBL", "TBLID"));
+        ColumnarFormatVersion columnarFormatVersion = ColumnarFormatVersion.V3;
+        Path path = new Path(
+                "/home/store/database/Fact/Part0/Segment_0/part-0-0_batchno0-0-1502197243476.carbondata");
```

Comment:
This path "/home/store/database/Fact/Part0/Segment_0/part-0-0_batchno0-0-1502197243476.carbondata" is not exist, why use it ?

Reply:
@chenliang613 This is just a dummy path, which is being used to create an object of Input Split. If you want we can rename it to mockPath.